libs/deal-ticket/src/components/deal-ticket/type-selector.tsx

```diff
@@ -1,32 +1,118 @@
 import {
-  FormGroup,
+  Icon,
   InputError,
   SimpleGrid,
   Tooltip,
+  TradingDropdownContent,
+  TradingDropdownItemIndicator,
+  TradingDropdownRadioGroup,
+  TradingDropdownRadioItem,
 } from '@vegaprotocol/ui-toolkit';
 import { t } from '@vegaprotocol/i18n';
-import * as Schema from '@vegaprotocol/types';
-import { Toggle } from '@vegaprotocol/ui-toolkit';
-import type { Market, MarketData } from '@vegaprotocol/markets';
+import type { Market, StaticMarketData } from '@vegaprotocol/markets';
 import { compileGridData } from '../trading-mode-tooltip';
 import { MarketModeValidationType } from '../../constants';
+import { DealTicketType } from '../../hooks/use-type-store';
+import * as RadioGroup from '@radix-ui/react-radio-group';
+import * as DropdownMenu from '@radix-ui/react-dropdown-menu';
+import classNames from 'classnames';
+import { FLAGS } from '@vegaprotocol/environment';
 
 interface TypeSelectorProps {
-  value: Schema.OrderType;
-  onSelect: (type: Schema.OrderType) => void;
+  value: DealTicketType;
+  onValueChange: (type: DealTicketType) => void;
   market: Market;
-  marketData: MarketData;
+  marketData: StaticMarketData;
   errorMessage?: string;
 }
 
 const toggles = [
-  { label: t('Limit'), value: Schema.OrderType.TYPE_LIMIT },
-  { label: t('Market'), value: Schema.OrderType.TYPE_MARKET },
+  { label: t('Limit'), value: DealTicketType.Limit },
+  { label: t('Market'), value: DealTicketType.Market },
 ];
+const options = [
+  { label: t('Stop Limit'), value: DealTicketType.StopLimit },
+  { label: t('Stop Market'), value: DealTicketType.StopMarket },
+];
+
+export const TypeToggle = ({
+  value,
+  onValueChange,
+}: Pick<TypeSelectorProps, 'onValueChange' | 'value'>) => {
+  const selectedOption = options.find((t) => t.value === value);
+  return (
+    <RadioGroup.Root
+      name="order-type"
+      className="mb-2 grid grid-cols-3 h-8 leading-8 font-alpha text-sm"
+      value={value}
+      onValueChange={onValueChange}
+    >
+      {toggles.map(({ label, value: itemValue }) => (
+        <RadioGroup.Item
+          value={itemValue}
+          key={itemValue}
+          id={`order-type-${itemValue}`}
+          data-testid={`order-type-${itemValue}`}
+          asChild
+        >
+          <button
+            className={classNames('rounded', {
+              'bg-vega-clight-500 dark:bg-vega-cdark-500': value === itemValue,
+            })}
+          >
+            {label}
+          </button>
+        </RadioGroup.Item>
+      ))}
+      {FLAGS.STOP_ORDERS && (
+        <DropdownMenu.Root>
```

Comment:
Can we not just use a more 'default' TradingDropdown

TradingDropdown lets you pass any trigger element now so that the styleing of the trigger button is not constricted